File: June-LeetCoding-Challenge/06-Queue-Reconstruction-By-Height/Queue-Reconstruction-By-Height.py
class Solution1:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        if len(people)<=1:
            return people
        people.sort(reverse=True)
        queue = []
        tmp = [people[0]]
        for i in range(1,len(people)):
            if people[i-1][0]==people[i][0]:
                tmp.append(people[i])
            else:
                queue+=sorted(tmp)
                tmp = [people[i]]
        queue+=sorted(tmp)
        res = [queue[0]]
        i = 1
        while queue[i][0]==queue[0][0] and i<len(queue):
            res.append(queue[i])
            i+=1
        while i<len(queue):
            res.insert(queue[i][1],queue[i])
            i+=1
        return res

File: June-LeetCoding-Challenge/06-Queue-Reconstruction-By-Height/Queue-Reconstruction-By-Height.cpp
bool compare(vector<int>& p, vector<int>& q) {
        return (p[0] > q[0]) || (p[0] == q[0] && p[1] < q[1]);
}

class Solution {
public:
    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
        sort(people.begin(), people.end(), compare);
        vector<vector<int>> ans;
        for(int i = 0; i < people.size(); i++)
            ans.insert(ans.begin() + people[i][1], people[i]);
        return ans;
    }
};